File: Data Structures & Algorithms/top-k-elements-in-list/submission-13.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        res = []
        counts = {}

        for num in nums:
            if num in counts:
                counts[num] += 1
            else:
                counts[num] = 1
        
        max_elem = nums[0]
        
        for i in range(k):
            for key in counts:
                max_elem = key if counts[key] > counts[max_elem] else max_elem
            res.append(max_elem)
            counts[max_elem] = 0
        
        return res

File: Data Structures & Algorithms/permutations/submission-13.java
class Solution {
    public List<List<Integer>> permute(int[] nums) {
        List<List<Integer>> perms = Arrays.asList(new ArrayList<>());

        for (int num : nums) {
            List<List<Integer>> res = new ArrayList<>();
            for (List<Integer> p : perms) {
                for (int i = 0; i <= p.size(); i++) {
                    List<Integer> p_copy = new ArrayList<>(p);
                    p_copy.add(i, num);
                    res.add(p_copy);
                }
            }
            perms = res;
        }

        return perms;
    }
}
